ovos_PHAL_plugin_oauth/__init__.py

```diff
@@ -26,18 +33,27 @@ class OAuthPlugin(PHALPlugin):
 
     def __init__(self, bus=None, config=None):
         super().__init__(bus=bus, name="ovos-PHAL-plugin-oauth", config=config)
```

Comment:
moving this line to the bottom of `__init__` method should fix your issues and avoid the need for a threading.Event completely

Reply:
Moving ``` super().__init__(bus=bus, name="ovos-PHAL-plugin-oauth", config=config) ``` to the bottom of __init__ causes other issues:

``` Traceback (most recent call last):
  File "/home/aix/ovos/ovos-core/.venv/lib/python3.10/site-packages/ovos_PHAL/service.py", line 74, in load_plugins
    self.drivers[name] = plug(bus=self.bus, config=config)
  File "/home/aix/ovos/ovos-core/.venv/lib/python3.10/site-packages/ovos_PHAL_plugin_oauth/__init__.py", line 83, in __init__
    self.bus.on("oauth.register", self.handle_oauth_register)
AttributeError: 'OAuthPlugin' object has no attribute 'bus' ```

Reply:
meant to make a PR but accidentally commited to branch directly, sorry!

changed this

Reply:
thanks, seems to have fixed it.